test/test_distributions.py

```diff
@@ -4905,8 +4905,8 @@ def f(*values):
 
             # check on different data
             values, sample = self._perturb(Dist, keys, values, sample)
-            expected = f(*values)
-            actual = traced_f(*values)
+            expected = f(*values).clone()
+            actual = traced_f(*values).clone()
```

Comment:
Why did this have to change?

Reply:
The next line writes into a broadcasted array:
`expected[expected == float('inf')] = 0.`

Reply:
This seems BC-breaking. Even if `expected` is a broadcasted array, `expected[expected == float('inf')] = 0.` produces a correct result because it is setting the right elements to 0

Reply:
Note that this is a `masked_fill_` which is exactly the operation in the original issue. To preserve the correct uses, like
```python
expected[expected == float('inf')] = 0.
```
then we could inspect the mask along the broadcasted dimensions and only allow the fill if the mask is constant along broadcasted dimensions. Maybe deprecating that behavior as well.